# Biscuit WiFi STA userspace cleanup for CM12
# Apply from workspace/cm12 with: patch -p1 < ../patches/cm12-biscuit-wifi-sta-userspace.patch

--- a/vendor/amazon/biscuit/proprietary/etc/wifi/wpa_supplicant.conf
+++ b/vendor/amazon/biscuit/proprietary/etc/wifi/wpa_supplicant.conf
@@ -7,6 +7,3 @@
 serial_number=2.0
 device_type=10-0050F204-5
 os_version=01020300
-config_methods=display push_button keypad
-p2p_no_group_iface=1
-driver_param=use_p2p_group_interface=1
--- a/vendor/amazon/biscuit/proprietary/etc/wifi/wpa_supplicant_overlay.conf
+++ b/vendor/amazon/biscuit/proprietary/etc/wifi/wpa_supplicant_overlay.conf
@@ -1,3 +1 @@
-p2p_no_group_iface=1
-driver_param=use_p2p_group_interface=1
 wowlan_triggers=disconnect
--- a/external/wpa_supplicant_8/wpa_supplicant/ctrl_iface.c
+++ b/external/wpa_supplicant_8/wpa_supplicant/ctrl_iface.c
@@ -5704,6 +5704,7 @@
 
 	ret = wpa_drv_driver_cmd(wpa_s, cmd, buf, buflen);
 	if (ret == 0) {
+#ifdef CONFIG_P2P
 		if (os_strncasecmp(cmd, "COUNTRY", 7) == 0) {
 			struct p2p_data *p2p = wpa_s->global->p2p;
 			if (p2p) {
@@ -5714,6 +5715,7 @@
 				p2p_set_country(p2p, country);
 			}
 		}
+#endif /* CONFIG_P2P */
 		ret = os_snprintf(buf, buflen, "%s\n", "OK");
 	}
 	return ret;
--- a/frameworks/opt/net/wifi/service/java/com/android/server/wifi/WifiStateMachine.java
+++ b/frameworks/opt/net/wifi/service/java/com/android/server/wifi/WifiStateMachine.java
@@ -5220,7 +5220,9 @@
                     if (mWifiConfigStore.associatedPartialScanPeriodMilli < 0) {
                         mWifiConfigStore.associatedPartialScanPeriodMilli = 0;
                     }
-                    initializeWpsDetails();
+                    if (mP2pSupported) {
+                        initializeWpsDetails();
+                    }
 
                     mIbssSupported = mWifiNative.getModeCapability("IBSS");
 
